core/src/services/fs/writer.rs

```diff
@@ -101,3 +108,128 @@ impl oio::BlockingWrite for FsWriter<std::fs::File> {
         Ok(())
     }
 }
+
+impl oio::PositionWrite for FsWriter<tokio::fs::File> {
+    #[cfg(not(target_os = "windows"))]
+    async fn write_all_at(&self, offset: u64, buf: Buffer) -> Result<()> {
+        let f = self.f.as_ref().expect("FsWriter must be initialized");
+
+        let f = f
+            .try_clone()
+            .await
+            .map_err(new_std_io_error)?
+            .into_std()
+            .await;
+
+        tokio::task::spawn_blocking(move || {
+            f.write_all_at(&buf.to_bytes(), offset)
```

Comment:
Please don't use `to_bytes` whenever possible.

Reply:
what method should i use?

Reply:
Use `buf.chunk()` and `buf.advance()`. I think we can implement a `write_at` under different target so we don't need to remove the `write` impl.

Reply:
Like this?